sdk/identity/identity/samples/AzureIdentityExamples.md

```diff
@@ -0,0 +1,347 @@
+- [Authenticating a service principal with a client secret](#authenticating-a-service-principal-with-a-client-secret)
+- [Authenticating a service principal with a client certificate](#authenticating-a-service-principal-with-a-client-certificate)
+- [Authenticating a service principal with environment credentials](#authenticating-a-service-principal-with-environment-credentials)
+- [Authenticating a user account with device code flow](#authenticating-a-user-account-with-device-code-flow)
+- [Authenticating a user account with username and password](#authenticating-a-user-account-with-username-and-password)
+- [Authenticating a user account interactively in the browser](#authenticating-a-user-account-interactively-in-the-browser)
+- [Authenticating a user account with auth code flow](#authenticating-a-user-account-with-auth-code-flow)
+- [Authenticating a user account with Azure CLI](#authenticating-a-user-account-with-azure-cli)
+- [Authenticating a user account with Visual Studio Code](#authenticating-a-user-account-with-visual-studio-code)'
+- [Authenticating in Azure with managed identity](#authenticating-in-azure-with-managed-identity)
+- [Chaining credentials](#chaining-credentials)
+- [Authenticating with Azure Stack using Azure Identity](#authenticating-with-azure-stack-using-azure-identity)
+
+## Introduction
+
+Authenticating your application, users, and principals is an integral part of working with the Azure Client Libraries. The Azure Identity library provides multiple ways to authenticate, each with a flexible configuration that covers most scenarios. In this document we will go over some of these scenarios and provide small examples that can be used as a starting point for your needs.
+
+## Authenticating with `DefaultAzureCredential`
+
+This example demonstrates authenticating the `SecretClient` from the [@azure/keyvault-secrets][secrets_client_library] client library using the `DefaultAzureCredential`. There's also [a runnable sample](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/identity/identity/samples/typescript/src/defaultAzureCredential.ts) to create a Key Vault key client you can copy-paste. The `DefaultAzureCredential` makes for a terrific starting point as it provides sane defaults with minimal configuration and chains multiple credentials together. While you may outgrow it eventually, it is a sensible first choice for most scenarios where the application is intended to ultimately be run in the Azure Cloud.
+
+```ts
+/**
+ * The default credential first checks environment variables for configuration.
+ * If environment configuration is incomplete, it will try managed identity.
+ */
+function withDefaultAzureCredential() {
+  const credential = new DefaultAzureCredential();
+  const client = new SecretClient(`https://key-vault-name.vault.azure.net`, credential);
+}
+```
+
+### Authenticating a user assigned managed identity with `DefaultAzureCredential`
```

Comment:
Is this the only customization of DefaultAzureCredential that is worth calling out?

Reply:
I don't have enough context to answer that question unfortunately but happy to callout anything y'all feel is important. I can tell that managedIdentityClientId is a common enough scenario to be called out but I don't know if we necessarily need to call out every customization we offer (tenantId and authorityHost - which _is_ covered in Azure Stack example below) @sadasant ?

Reply:
`managedIdentityClientId` is the only customization we have on DAC at the moment: https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/identity/identity/src/credentials/defaultAzureCredential.ts#L51-L54